### DISCUSSION TITLE
DeprecationWarning on convertor from XML to YAML

### DISCUSSION THREAD

**Discussion by florian-vuillemot** [shadow/shadow#971] [Q&A] [answered]
**State your question concisely**
I ran the `src/tools/convert.py` script with Python3.8, my `LXML` package version is `4.5.2` and I obtained the following warning: 
```
convert.py:138: DeprecationWarning: This method will be removed in future versions.  Use 'list(elem)' or iteration over elem instead.
  **xml_nodes_to_dict(root.getchildren())
```

**Describe your current understanding of Shadow as it relates to your question**
We are doing some work to convert XML to YAML configuration files to be more readable. But today, as Shadow is not allowed to read this kind of file, we have to reconvert their files to XML before using it.

**Describe which parts you are unsure about**
Should we update this script to remove this warning and handle the next version of the `LXML` package or should we consider that we will replace the XML loader by a YAML loader beforehand ?

**Comment by stevenengler** [comment, marked as answer]
It sounds like a small change, so I think it would be good to fix the deprecation notice. Even after Shadow supports yaml, other people may want to convert their own xml configs to yaml, so we'll want to keep this script around for awhile.

**Comment by florian-vuillemot** [comment]
Ok, do you think I can handle that update in the https://github.com/shadow/shadow/pull/972 ?

**Comment by stevenengler** [comment]
If it's just a couple of lines and doesn't change the script's functionality, then feel free to update it in #972 as its own commit.
